### PR TITLE
Unit trading

### PR DESCRIPTION
**Pull request by GauthamN**
Trading unit tests
#546

### CODE REVIEW

File: src/contracts/fund/trading/unit/trading.test.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      takesCustodyMasks,
+    ]),
+  ).rejects.toThrow(errorMessage);
+});
+
+test('returnToVault sends back token balances to the vault', async () => {
+  const tokenQuantity = new BigInteger(10 ** 20);
+
+  await shared.mln.methods
+    .transfer(shared.trading.options.address, `${tokenQuantity}`)
```
Comment (schmidsi) [inline src/contracts/fund/trading/unit/trading.test.ts]: You could use the `contracts/dependencies/token/transactions/` here.

File: src/utils/constants/orderSignatures.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+function joinSignature(name, argTypes = []) {
+  return `${name}(${argTypes.join(',')})`;
+}
+
+function abiEncode(joinedSignature) {
+  return Abi.encodeFunctionSignature(joinedSignature);
+}
+
+export const makeOrderSignature = joinSignature('makeOrder', [
```
Comment (schmidsi) [inline src/utils/constants/orderSignatures.ts]: Actually I don't understand the value of this. We should derive these signatures directly from the ABI, or just provide strings. Why make an array and join it directly?

File: src/utils/helpers/randomAddress.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import * as Web3Utils from 'web3-utils';
 import { Address } from '~/utils/types';
 
-export const randomAddress = () => new Address(Web3Utils.randomHex(20));
+export const randomAddress = () =>
+  new Address(Web3Utils.randomHex(20)).toLowerCase();
```
Comment (schmidsi) [inline src/utils/helpers/randomAddress.ts]: Why `toLowerCase`? I think its good practice to try to always use the check-summed addresses. But I also know that here and there that leads to problems (for example comparing).

Comment (GauthamN) [reply src/utils/helpers/randomAddress.ts]: Changed it back

File: src/utils/constants/orderSignatures.ts (hunk 2 above)
Comment (GauthamN) [reply src/utils/constants/orderSignatures.ts]: We need signature string along with bytes. I didn't find web3 function that returns signature string by taking function name as parameter. And joining as a separate step is just for better readability of signature string.

**Pull request merged**